datafusion/sqllogictest/test_files/aggregate.slt

```diff
@@ -2744,28 +2735,84 @@ SELECT ARRAY_AGG([1])
 ----
 [[1]]
 
-# test_approx_percentile_cont_decimal_support
-query TI
-SELECT c1, approx_percentile_cont(c2, cast(0.85 as decimal(10,2))) apc FROM aggregate_test_100 GROUP BY 1 ORDER BY 1
+# test array_agg with no row qualified
+statement ok
+create table t(a int, b float, c bigint) as values (1, 1.2, 2);
+
+query ?
+select array_agg(a) from t where a > 2;
 ----
-a 4
-b 5
-c 4
-d 4
-e 4
+NULL
 
+query ?
+select array_agg(b) from t where b > 3.1;
+----
+NULL
 
-# array_agg_zero
 query ?
-SELECT ARRAY_AGG([]);
+select array_agg(c) from t where c > 3;
 ----
-[[]]
+NULL
 
-# array_agg_one
+query ?I
+select array_agg(c), count(1) from t where c > 3;
+----
+NULL 0
+
+# returns 0 rows if group by is applied
 query ?
-SELECT ARRAY_AGG([1]);
+select array_agg(a) from t where a > 3 group by a;
 ----
-[[1]]
+
+query ?I
+select array_agg(a), count(1) from t where a > 3 group by a;
+----
+
+# TODO: Expect NULL, got empty list
+query ?
+select array_agg(distinct a) from t where a > 3;
```

Comment:
This is the only query that has unexpected result.

Reply:
Thanks @jayzhan211 for calling this out!

Given there are no rows with `a > 3`, the aggregation input relation `from t where a > 3` is empty, so this should produce same result as array_agg from empty table, ie `NULL`. currently it shows `[]` as a result.

Reply:
Maybe it is related to a multi-stage grouping (where the first phase outputs a NULL and then the second phase gets that NULL and interprets it as `[]` 🤔

Reply:
The results look good now